src/proof/drat/drat_proof.cpp

```diff
@@ -0,0 +1,123 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2022 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * Declares C++ types that represent a DRAT proof.
+ * Defines serialization for these types.
+ *
+ * You can find an introduction to DRAT in the drat-trim paper:
+ *    http://www.cs.utexas.edu/~marijn/publications/drat-trim.pdf
+ */
+
+#include "proof/drat/drat_proof.h"
+
+#include <algorithm>
+#include <bitset>
+#include <iostream>
+
+namespace cvc5::internal {
+namespace proof {
+
+// DratInstruction implementation
+DratInstruction::DratInstruction(DratInstructionKind kind,
+                                 prop::SatClause clause)
+    : d_kind(kind), d_clause(clause)
+{
+  // All intialized
+}
+
+// DratProof implementation
+
+DratProof::DratProof() : d_instructions() {}
+
+std::vector<std::string> splitString(std::string s, std::string splitter)
+{
+  std::vector<std::string> lines;
+  size_t pos = 0;
+  std::string token;
+  while ((pos = s.find(splitter)) != std::string::npos)
+  {
+    token = s.substr(0, pos);
+    s.erase(0, pos + splitter.length());
+    if (token.length() > 0)
+    {
+      lines.push_back(token);
+    }
+  }
+  token = s.substr(0, pos);
+  if (token.length() > 0)
+  {
+    lines.push_back(token);
+  }
+  return lines;
+}
```

Comment:
This function looks overly complicated for what it should do. `splitter` is always a character. In this case you could use `std::getline` to split the string as it is done here: https://github.com/murxla/murxla/blob/main/src/util.cpp#L631

```
std::vector<std::string>
split(const std::string& s, const char delim)
{
  std::stringstream ss(s);
  std::string token;
  std::vector<std::string> res;
  while (std::getline(ss, token, delim))
  {
    res.push_back(token);
  }
  return res;
}
```